module/VuFind/src/VuFind/Controller/CombinedController.php

```diff
@@ -325,15 +335,18 @@ protected function adjustQueryForSettings($settings, $searchType = null)
         // Override the search type:
         $query->type = $searchType;
 
-        // Always leave noresults active (useful for 0-hit searches) and
-        // side inactive (no room to display) but display or hide top based
-        // on include_recommendations setting.
-        if ($settings['include_recommendations'] ?? false) {
-            $query->noRecommend = 'side';
-            if (is_array($settings['include_recommendations'])) {
-                $query->recommendOverride
-                    = ['top' => $settings['include_recommendations']];
+        // Always leave noresults active (useful for 0-hit searches).
+        // Display or hide top based on include_recommendations setting.
+        // Display or hide top based on include_recommendations_side setting.
+        if (($settings['include_recommendations'] ?? false) || ($settings['include_recommendations_side'] ?? false)) {
+            $recommendOverride = [];
+            if (is_array($settings['include_recommendations'] ?? null)) {
+                $recommendOverride['top'] = $settings['include_recommendations'];
+            }
```

Comment:
Do we need else clauses to set the appropriate noRecommend options for missing settings? I confess that I can't exactly remember how this all works, so maybe it's not necessary, but just looking at the refactored code, it seems like in the existing code, if we only have top recommendations, we set noRecommend to side, so I wonder if we still need to do that (and also support setting noRecommend to top if we only have side settings).

Reply:
Yes I think you're right.  I rewrote this logic to (I think) handle `noRecommend` correctly.